### PR TITLE
Nested `with` blocks

### PR DESCRIPTION
**Pull request by awr1**
Just a small addition to `with` that I felt would be useful for complicated objects.

### CODE REVIEW

File: lib/std/with.nim [hunk 1]
```diff
@@ -35,5 +35,15 @@ macro with*(arg: typed; calls: varargs[untyped]): untyped =
...
+    with foo:
+      bar = 2
+      with qux:
+        baz = 3
+    doAssert foo.bar == 2
+    doAssert foo.qux.baz == 3
+
   result = newNimNode(nnkStmtList, arg)
   underscoredCalls(result, calls, arg)
+  echo result.astGenRepr
```
Comment (rockcavera) [inline lib/std/with.nim:49]: Is there any reason to keep this?

Comment (ringabout) [reply lib/std/with.nim:49]: removed here https://github.com/nim-lang/Nim/pull/22047

Comment (rockcavera) [reply lib/std/with.nim:49]: I had not seen this PR

Comment (awr1) [reply lib/std/with.nim:49]: @ringabout @rockcavera Oops, my bad for when I was testing. Thanks for fixing!

**Pull request merged**